src/main/scala/io/citrine/lolo/Model.scala

```diff
@@ -9,7 +11,21 @@ trait Model[+T <: PredictionResult[Any]] extends Serializable {
     * Apply the model to a seq of inputs
     *
     * @param inputs to apply the model to
-    * @return a predictionresult which includes, at least, the expected outputs
+    * @return a PredictionResult which includes, at least, the expected outputs
     */
   def transform(inputs: Seq[Vector[Any]]): T
+
+  /**
+    * Compute Shapley feature attributions for a given input in this node's subtree
+    *
+    * @param input for which to compute feature attributions.
+    * @param omitFeatures feature indices to omit in computing Shapley values
+    * @return array of vector-valued attributions for each feature
+    *         One DenseVector[Double] per feature, each of length equal to the output dimension.
+    */
+  def shapley(
```

Comment:
Why is `shapley` defined at such a high level? It doesn't seem like something that would extend to all models.

Reply:
It does extend to all models. There's a simple closed form available for linear regression, for example, and a decent kernel approximation for general black-box models.